### ISSUE TITLE
Thrift 0.9.1

### ISSUE THREAD

**Issue by aurrelhebert** [cityzendata/warp10-platform#215]
Thrift 0.9.1 broke with Ubuntu 6.3.0-12.

**Comment by hbs** [comment]
???

**Comment by morind** [comment]
Yes Aurélien some issues during install of thrift 0.9.1 with recent Ubuntu releases..

1. 

```
g++-6 -DHAVE_CONFIG_H -I. -I../.. -I../../lib/cpp/src/thrift  -I./src -I/usr/local/opt/openssl/include -Wall -g -O2 -MT thrift-t_java_generator.o -MD -MP -MF .deps/thrift-t_java_generator.Tpo -c -o thrift-t_java_generator.o `test -f 'src/generate/t_java_generator.cc' || echo './'`src/generate/t_java_generator.cc
src/generate/t_java_generator.cc: In member function ‘void t_java_generator::generate_process_async_function(t_service*, t_function*)’:
src/generate/t_java_generator.cc:2830:14: error: operands to ?: have different types ‘bool’ and ‘std::basic_ostream<char>’
        **first ? first = false : indent(f_service_) << "else ";**
        ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Makefile:700: recipe for target 'thrift-t_java_generator.o' failed
```
**Workaround:** use clang
Source: [https://issues.apache.org/jira/browse/AURORA-1727](https://issues.apache.org/jira/browse/AURORA-1727)

2.

```
make[3]: Entering directory '/var/tmp/thrift-0.9.1-src/test/cpp'
Makefile:832: warning: overriding recipe for target 'gen-cpp/ThriftTest.cpp'
Makefile:829: warning: ignoring old recipe for target 'gen-cpp/ThriftTest.cpp'
/bin/bash ../../libtool --tag=CXX   --mode=link clang++ -Wall -g -O2 -L/usr/lib   -o libtestgencpp.la  ThriftTest_constants.lo ThriftTest_types.lo ../../lib/cpp/libthrift.la -lssl -lcrypto -lrt -lpthread 
libtool: link: ar cru .libs/libtestgencpp.a .libs/ThriftTest_constants.o .libs/ThriftTest_types.o 
ar: `u' modifier ignored since `D' is the default (see `U')
ar: .libs/ThriftTest_constants.o: No such file or directory
Makefile:446: recipe for target 'libtestgencpp.la' failed
```
**Workaround:** use ./configure --without-tests 
Source: [https://stackoverflow.com/questions/18643642/libtool-error-building-thrift-0-9-1-on-ubuntu-13-04](https://stackoverflow.com/questions/18643642/libtool-error-building-thrift-0-9-1-on-ubuntu-13-04)

**To sum up:**
- sudo apt install clang
- export CC=clang; export CXX=clang++
- ./configure --without-python --without-ruby --without-tests --prefix=/var/tmp/thrift-0.9.1 CPPFLAGS='-I/usr/local/opt/openssl/include'
- make
- sudo make install

**Comment by aurrelhebert** [comment]
Thanks a lot for the detailed notes about the compilation error I encountered with Thrift! This fix works fine on my side. I also had to add ant which is required to have the java library.

**Comment by hbs** [comment]
This is an Ubuntu/Thrift issue, not a Warp 10 one, closing it.
